embedded-graphics/src/primitives/ellipse.rs

```diff
@@ -0,0 +1,512 @@
+            center: styled.primitive.center_2x(),
+            threshold,
+        }
+    }
+}
+
+impl<C> Iterator for StyledEllipseIterator<C>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        for point in &mut self.iter {
+            let inside_border = is_point_inside_ellipse(
+                self.inner_size_sq,
+                point * 2 - self.center,
+                self.threshold,
+            );
+
+            let color = if inside_border {
+                self.inner_color
+            } else {
+                self.outer_color
+            };
+
+            if let Some(color) = color {
+                return Some(Pixel(point, color));
+            }
+        }
+
+        None
+    }
+}
+
+impl<'a, C: 'a> Drawable<C> for &Styled<Ellipse, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    fn draw<D: DrawTarget<C>>(self, display: &mut D) -> Result<(), D::Error> {
+        display.draw_ellipse(self)
+    }
+}
+
+fn compute_threshold(size: Size, diameter: u32) -> u32 {
+    let Size {
+        width: a,
+        height: b,
+    } = size;
+
+    // Special case for circles, where width and height are equal
+    if a == b {
+        if diameter <= 4 {
+            diameter.pow(2) - diameter / 2
+        } else {
+            diameter.pow(2)
+        }
+    } else {
+        b * a
+    }
+}
+
+/// Uses the ellipse equation b^2 * x^2 + a^2 * y^2 - a^2 * b^2 to return a value signifying whether
+/// a given point lies inside (`true`) or outside (`false`) an ellipse centered around `(0, 0)` with
+/// width and height defined by the `size` parameter.
+fn is_point_inside_ellipse(size: Size, point: Point, threshold: u32) -> bool {
+    let Size {
+        width: a,
+        height: b,
+    } = size;
+
+    let Point { x, y } = point;
+
+    let x = x.pow(2) as u32;
+    let y = y.pow(2) as u32;
+
+    // Special case for circles, where width and height are equal
+    if a == b {
+        (x + y) < threshold
+    } else {
+        b * x + a * y < threshold
+    }
+}
+
+impl<'a, C> IntoIterator for &'a Styled<Ellipse, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+    type IntoIter = StyledEllipseIterator<C>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StyledEllipseIterator::new(self)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{
+        mock_display::MockDisplay, pixelcolor::BinaryColor, primitives::Circle,
+        style::PrimitiveStyleBuilder,
+    };
+
+    fn test_ellipse(size: Size, style: PrimitiveStyle<BinaryColor>, pattern: &[&str]) {
+        let mut display = MockDisplay::new();
+
+        Ellipse::new(Point::new(0, 0), size)
+            .into_styled(style)
+            .draw(&mut display)
+            .unwrap();
+
+        assert_eq!(display, MockDisplay::from_pattern(pattern));
+    }
+
+    fn test_circles(style: PrimitiveStyle<BinaryColor>) {
+        for diameter in 1..50 {
```

Comment:
Why did you change this to start from `1` instead of `0`? IMO diameter `0` should also match.

Reply:
Oops, that was a debugging artifact left over. It should indeed by `0..50`. Changed!

Reply:
You only changed this in `circles_points` and not in `test_circles`.